src/System.Private.TypeLoader/src/System/Reflection/Runtime/General/QHandles.NativeFormat.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// Collection of "qualified handle" tuples.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.Metadata.NativeFormat;
+using Internal.Runtime.TypeLoader;
+
+namespace System.Reflection.Runtime.General
+{
+
+    public partial struct QMethodDefinition
+    {
+        public QMethodDefinition(MetadataReader reader, MethodHandle handle)
+        {
+            _reader = reader;
+            _handle = ((Handle)handle).AsInt();
+        }
+
+        public MetadataReader NativeFormatReader { get { return _reader as MetadataReader; } }
```

Comment:
NativeFormatReader should assert that IsNativeFormat is true. (Similarly for other properties whose applicability is conditional upon the native/ecma discriminant.)